examples/3_simple_storage/simple_storage.go

```diff
@@ -0,0 +1,231 @@
+/*
+WHAT IS IN THIS EXAMPLE?
+
+Keybase has added an encrypted key-value store intended to support
+security-conscious bot development with persistent state. It is a
+place to store small bits of data that are
+  (1) encrypted for a team or user (via the user's implicit self-team: e.g.
+alice,alice),
+  (2) persistent across logins
+  (3) fast and durable.
+
+It supports putting, getting, listing, and deleting. There is also
+concurrency support, but check out 5_secret_storage for that. A team has many
+namespaces, a namespace has many entryKeys, and an entryKey has one current
+entryValue. Namespaces and entryKeys are in cleartext, and the Keybase client
+service will encrypt and sign the entryValue on the way in (as well as
+decrypt and verify on the way out) so keybase servers cannot see it or
+forge it.
+
+This example shows how you can use interact with the team
+encrypted key-value store using chat commands.
+*/
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat"
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/chat1"
+)
+
+func fail(msg string, args ...interface{}) {
+	fmt.Fprintf(os.Stderr, msg+"\n", args...)
+	os.Exit(3)
+}
+
+/*
+ Listens to chat messages of the form:
+    `!storage put <namespace> <key> <value> (<revision>)`
+    `!storage get <namespace> <key>`
+    `!storage delete <namespace> <key> (<revision>)`
+    `!storage list`  // list namespaces
+    `!storage list <namespace>`  // list entries in namespace
+*/
+func main() {
+	const MsgPrefix = "!storage"
+	const (
+		Help   = "help"
+		List   = "list"
+		Get    = "get"
+		Put    = "put"
+		Delete = "delete"
+	)
+
+	var kbLoc string
+	var kbc *kbchat.API
+	var err error
+
+	flag.StringVar(&kbLoc, "keybase", "keybase", "the location of the Keybase app")
+	flag.Parse()
+
+	fmt.Println("Simple kvstore storage bot now starting...")
+
+	if kbc, err = kbchat.Start(kbchat.RunOptions{KeybaseLocation: kbLoc}); err != nil {
+		fail("Error creating API: %s", err.Error())
+	}
+
+	sub, err := kbc.ListenForNewTextMessages()
+	if err != nil {
+		fail("Error listening: %s", err.Error())
+	}
+
+	fmt.Println("Now listening for new chat messages...")
```

Comment:
I think the example could be simplified by just showing the get/put/delete/list properties of the storage without having it driven by chat messages. No strong opinion here but think it may help someone new to the library understand the value of the system better.

Reply:
yeah - actually, i agree. i think that would make it much simpler and more straightforward.